google/cloud/bigtable/tests/instance_admin_emulator.cc

```diff
@@ -187,17 +195,21 @@ class InstanceAdminEmulator final
   grpc::Status ListClusters(grpc::ServerContext* context,
                             btadmin::ListClustersRequest const* request,
                             btadmin::ListClustersResponse* response) override {
-    std::string prefix = request->parent();
+    std::string instance_path =
+        request->parent().substr(0, request->parent().find_last_of("/"));
+    std::string prefix = request->parent() + "/clusters/";
 
-    // magical instance name "-" must return all clusters in the project
-    if (request->parent().back() == '-') {
-      for (auto const& kv : clusters_) {
-        *response->add_clusters() = kv.second;
+    for (auto const& instance : instances_) {
+      if (std::string::npos == instance.first.find(instance_path)) {
```

Comment:
This is still a bit confusing, can you explain what `instance_path` is supposed to be?  I *think* that is this the
`projects/<project_id>/instances` part of the instance name, and therefore any instance or cluster outside the project is ignored.  Assuming I got it right (and I am still not sure if this was your intent), I had to spend time thinking that through where I comment would have helped.

Reply:
Yes that was my intent
I will change it to more correct way